src/common/components/repositories-list/index.js

```diff
@@ -69,26 +70,28 @@ class RepositoriesList extends Component {
     }
   }
 
-  renderRow(snap) {
+  renderRow(snap, index) {
+    const { hasNoRegisteredNames, registerName, snapBuilds, authStore } = this.props;
     const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
 
     let latestBuild = null;
-    const snapBuilds = this.props.snapBuilds[fullName];
+    const currentSnapBuilds = snapBuilds[fullName];
 
-    if (snapBuilds && snapBuilds.success && snapBuilds.builds.length) {
-      latestBuild = snapBuilds.builds[0];
+    if (currentSnapBuilds && currentSnapBuilds.success && currentSnapBuilds.builds.length) {
+      latestBuild = currentSnapBuilds.builds[0];
     }
 
-    const registerNameStatus = this.props.registerName[fullName] || {};
+    const registerNameStatus = registerName[fullName] || {};
 
     return (
       <RepositoryRow
         key={ `repo_${fullName}` }
         snap={ snap }
         latestBuild={ latestBuild }
         fullName={ fullName }
-        authStore={ this.props.authStore }
+        authStore={ authStore }
         registerNameStatus={ registerNameStatus }
+        registerNameIsOpen={ !index && hasNoRegisteredNames }
```

Comment:
> Which can only be the first repo in the list because none of the repos have a name, right?

This was my reading of it. I think we should make the test slightly more specific, though:
`registerNameIsOpen={ index === 0 && hasNoRegisteredNames }`

`index === 0` could also be extracted into a variable named `isFirst` or `isFirstInList`.